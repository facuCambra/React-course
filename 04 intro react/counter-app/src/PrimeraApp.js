//import React, { Fragment } from 'react';   //imr tab importa react
import React from 'react';
import PropTypes from 'prop-types';
;


//functional components 

const PrimeraApp = ( {saludo = 'Hello', subtitulo}) => {
   

    console.log(saludo);
    return(
  
    <>
        <h1>{saludo}</h1>
        
    </>); 
}

PrimeraApp.defaultProps = {
    subtitulo: 'Defecto'
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

export default PrimeraApp;  //exp tab es el shortcut
