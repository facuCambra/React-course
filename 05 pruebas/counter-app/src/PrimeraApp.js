//import React, { Fragment } from 'react';   //imr tab importa react
import React from 'react';
import PropTypes from 'prop-types';
;


//functional components 

const PrimeraApp = ( {saludo = 'Hello', subtitulo}) => {
   

    //console.log(saludo);
    return(
  
    <>
        <h1>{saludo}</h1>
        
        <p> {subtitulo}</p>
    </>); 
}

PrimeraApp.defaultProps = {
    subtitulo: 'Sub'
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

export default PrimeraApp;  //exp tab es el shortcut
