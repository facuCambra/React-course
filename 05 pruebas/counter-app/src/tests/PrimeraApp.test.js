//import { render } from "@testing-library/react";
//import { render } from "react-dom";
import PrimeraApp from "../PrimeraApp";
import React from 'react';
import '@testing-library/jest-dom' //para las ayudas
import {shallow} from 'enzyme';

// import '@testing-library/jest-dom/extend-expect';
// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// Enzyme.configure({ adapter: new Adapter() });



describe('Pruebas en PrimeraApp', ()=>{
    // test('debe mostar el mensaje "Saludos"', () => { 
        
    //     const saludoo = 'Saludos';
    //     const {getByText} = render(<PrimeraApp saludo={saludoo} />) ; //producto renderizado de mi componente
    //     expect( getByText(saludoo) ).toBeInTheDocument();
    // })

    test('Debe de mostrar <PrimeraApp/> correctamente', () => {
        const saludo = 'Buenas';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>); //permite simular varias cosas
        
        expect( wrapper).toMatchSnapshot(); //el snapshot
    
    })

    test('debe de mostarar el subtitulo enviado por props', () => { 
        const saludo = 'Saludooo';
        const subtitulo = 'Saludooo2';

        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text().trim();  //busca un parrafo (trim para sacar espacio al principio)
                                                      // puedo buscar por clase con punto o id #
        expect(textoParrafo).toBe(subtitulo);
     })
})
