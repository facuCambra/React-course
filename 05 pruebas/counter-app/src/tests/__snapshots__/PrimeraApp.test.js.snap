// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe de mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Buenas
  </h1>
  <p>
     
    Sub
  </p>
</Fragment>
`;
